### PR TITLE
Add documentation about `ActiveModel::Errors#[]` method to describe how it works [ci skip]

### PR DESCRIPTION
**Pull request by meinac**
Please see the conversation on #23468.

### CODE REVIEW

File: activemodel/lib/active_model/errors.rb [hunk 1]
```diff
@@ -160,6 +160,15 @@ def delete(key)
     #
     #   person.errors[:name]  # => ["cannot be nil"]
     #   person.errors['name'] # => ["cannot be nil"]
+    #
+    # Note that, if you try to get errors of an attribute which has
+    # any errors associated with it, this method will instantiate
```
Comment (maclover7) [inline activemodel/lib/active_model/errors.rb]: an attributes which has no errors

File: activemodel/lib/active_model/errors.rb [hunk 2]
```diff
@@ -147,9 +147,9 @@ def set(key, value)
 
     # Delete messages for +key+. Returns the deleted messages.
     #
-    #   person.errors[:name]    # => ["cannot be nil"]
+    #   person.errors[:name]        # => ["cannot be nil"]
```
Comment (maclover7) [inline activemodel/lib/active_model/errors.rb]: why do we need the extra spaces?

Comment (meinac) [inline activemodel/lib/active_model/errors.rb]: because as far as I know, this is a convention. All output messages needs to be aligned.

**Comment by meinac** [comment]
I've changed the documentation message in direction shown by @maclover7.

File: activemodel/lib/active_model/errors.rb (hunk 2 above)
Comment (sgrif) [inline activemodel/lib/active_model/errors.rb]: This is unrelated to the change you're actually making, and is just cosmetic git churn. If you want to token align the docs you're adding, fine, but please don't change unrelated docs.

Comment (meinac) [inline activemodel/lib/active_model/errors.rb]: ok, I understand. Just removed this change.
